PWGHF/TableProducer/candidateCreatorDstar.cxx

```diff
@@ -13,65 +13,415 @@
+        primaryVertex.setSigmaZ2(rowTrackIndexDstar.pvRefitSigmaZ2());
+        covMatrixPV = primaryVertex.getCov(); /// Here covMatrixPV Updated!
+      }
+      registry.fill(HIST("Refit/hCovPVXX"), covMatrixPV[0]);
+      registry.fill(HIST("Refit/hCovPVYY"), covMatrixPV[2]);
+      registry.fill(HIST("Refit/hCovPVXZ"), covMatrixPV[3]);
+      registry.fill(HIST("Refit/hCovPVZZ"), covMatrixPV[5]);
+
+      // get track impact parameters
+      o2::dataformats::DCA impactParameter0; // GPUROOTCartesianFwd.h
+      o2::dataformats::DCA impactParameter1;
+      // Propagating D0 prongs to DCA
+      trackD0ProngParVar0.propagateToDCA(primaryVertex, bz, &impactParameter0);
+      trackD0ProngParVar1.propagateToDCA(primaryVertex, bz, &impactParameter1);
+
+      // Propagating Soft Pi to DCA
+      o2::dataformats::DCA impactParameterPi;
+      trackPiParVar.propagateToDCA(primaryVertex, bz, &impactParameterPi);
+      registry.fill(HIST("QA/hDcaXYProngsD0"), trackD0Prong0.pt(), impactParameter0.getY() * cmToMicrometers);
+      registry.fill(HIST("QA/hDcaXYProngsD0"), trackD0Prong1.pt(), impactParameter1.getY() * cmToMicrometers);
+      registry.fill(HIST("QA/hDcaZProngsD0"), trackD0Prong0.pt(), impactParameter0.getZ() * cmToMicrometers);
+      registry.fill(HIST("QA/hDcaZProngsD0"), trackD0Prong1.pt(), impactParameter1.getZ() * cmToMicrometers);
+
+      registry.fill(HIST("QA/hDCAXYPi"), trackPi.pt(), impactParameterPi.getY() * cmToMicrometers);
+      registry.fill(HIST("QA/hDCAZPi"), trackPi.pt(), impactParameterPi.getZ() * cmToMicrometers);
+
+      // get uncertainty of the decay length
+      double phi, theta;
+      getPointDirection(std::array{primaryVertex.getX(), primaryVertex.getY(), primaryVertex.getZ()}, secondaryVertex, phi, theta);
+      // Calculates the XX element of a XYZ covariance matrix after rotation of the coordinate system by phi around the z-axis and by minus theta around the new y-axis.
+      auto errorDecayLength = std::sqrt(getRotatedCovMatrixXX(covMatrixPV, phi, theta) + getRotatedCovMatrixXX(covMatrixPCA, phi, theta));
+      auto errorDecayLengthXY = std::sqrt(getRotatedCovMatrixXX(covMatrixPV, phi, 0.) + getRotatedCovMatrixXX(covMatrixPCA, phi, 0.));
+
+      // Calculation of kinematics for inv mass
       auto pVecD0 = RecoDecay::pVec(pVecD0Prong0, pVecD0Prong1);
+      // D0 pt vector
+      auto pxD0 = pVecD0.at(0);
+      auto pyD0 = pVecD0.at(1);
+      auto ptVecD0 = std::array{pxD0, pyD0};
+      // D0 pt magnitude
+      auto ptD0 = RecoDecay::pt(ptVecD0);
+
+      // Soft pi momentum vector
+      std::array<float, 3> pVecSoftPi;
+      trackPiParVar.getPxPyPzGlo(pVecSoftPi);
+
+      // Dstar momentum
+      auto pVecDstar = RecoDecay::pVec(pVecD0, pVecSoftPi);
+      auto pxDstar = pVecDstar.at(0);
+      auto pyDstar = pVecDstar.at(1);
+      // Dstar ptVec
+      std::array<float, 2> ptVecDstar{pxDstar, pyDstar};
+      // DStar pt magnitude
+      auto ptDstar = RecoDecay::pt(ptVecDstar);
```

Comment:
This entire thing is unnecessary.
```suggestion
      // D* pT magnitude
      auto ptDstar = RecoDecay::pt(pVecD0, pVecSoftPi);
```

Reply:
I think it is unnecessary for experts in coding but it was not for me at first sight because the data type of argument is not clear in the documentation of recoDeacy::pt(). I will make sure to remove what you suggested here.